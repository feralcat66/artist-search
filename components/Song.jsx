import React from 'react';
import PropTypes from 'prop-types';

const Song = ({ songTitle = 'Hello Darkness My Old Friend', songLyrics = 'No Lyrics Found' }) => (
  <>
    <h3>{songTitle}</h3>
    <p>{songLyrics}</p>
  </>
);

Song.propTypes = {
  songTitle: PropTypes.string.isRequired,
  songLyrics: PropTypes.string
};

export default Song;
