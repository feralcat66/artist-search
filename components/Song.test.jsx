import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song.jsx';

describe('Song component', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song songTitle="Hello Darkness My Old Friend" songLyrics="No Lyrics Found" />);
    expect(wrapper).toMatchSnapshot();
  });
});
