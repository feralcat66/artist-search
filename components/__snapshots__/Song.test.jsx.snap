// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component renders Song 1`] = `
<Fragment>
  <h3>
    Hello Darkness My Old Friend
  </h3>
  <p>
    No Lyrics Found
  </p>
</Fragment>
`;
